docs/developer/data_import.md

```diff
@@ -0,0 +1,48 @@
+
+The bottom portion of the Import tab acts as a file browser and selector for starting the Import process. The staged files get rendered by the [StagingAreaViewer](../../kbase-extension/static/kbase/js/widgets/narrative_core/upload/stagingAreaViewer.js) module. This viewer fetches the list of files from the Staging Service, updating every 30 seconds.
+
+## Data Import
+
+Data import is initiated by selecting one or more data files from the Import tab, then running the Importer app for that data type. When at least one file has a data type chosen and its checkbox selected, the `Import Selected` button will activate. Clicking that will close the slideout and open a Bulk Import cell and/or at least one App Cell with the relative file path entered. File paths are given as the relative path from the root of the Staging Area without the username - the username is assumed to be the root, and will be added at app run time.
+
+### Importer App Configuration
+
+The data import app selection is configured in [`staging_upload.json`](../../kbase-extension/static/kbase/config/staging_upload.json). This JSON file has the following top-level keys:
+* `dropdown_order`: this describes the order of the file selection dropdown. Each item here has an `id` field, which must be unique, and acts as the app descriptor (more in the `app_info` field below), and a `name` field. The `name` field is displayed to the user in the dropdown.
+* `bulk_import_types`: This array contains data type `id`s that should trigger a Bulk Import cell. If a user imports at least one file with a data type in this list, it'll map that file and import app into a bulk import cell.
+* `app_info`: This object has a set of app ids for keys that map onto info about each app. These keys are used as the app `id`s elsewhere in the configuration.
+
+Each `app_info` object has the following accepted keys:
+* `app_id`: the app to use for importing the file in the format `module_name/app_name` (e.g. `kb_uploadmethods/import_genbank_as_genome_from_staging`). This is the only required field.
+* `app_input_param`: this is the parameter for setting the file name in the app
+* `app_input_param_type`: describes what the `app_input_param` is expected to be. Accepts either `string` or `list` (list of strings)
+* `app_static_params`: an object with key-value pairs of other app parameters that should be preset when creating the app cell.
+* `app_output_param`: the app's parameter id for the created object
+* `app_output_suffix`: an automatic suffix to add to the data file name for a created object. E.g. if the file name is "e_coli.gbk", and the `app_output_suffix` is "_genome", then the automatically set output name will be "e_coli.gbk_genome"
+* `multiselect`: originally intended to denote files with multiple inputs, this is not currently implemented
```

Comment:
Isn't this also redundant to `allow_multiple` in spec.json?

Reply:
...ish. I have a foggy memory of why we put that in there, and it's not used. I wanted to match the doc to the file.

Reply:
This one sounds like a candidate for removal in a future PR to simplify things

Reply:
👍